liquid_node/commands.py

```diff
@@ -450,6 +474,110 @@ def deletecollection(name):
     purge_collection(name)
 
 
+def importcollection(name, path, method='copy'):
+    """Import a single collection extracted collection.
+    The path must contain:
+    1. The blobs folder, named `blobs`
+    2. The database folder, named `pg`, containing a subfolder called `data`.
+    3. the index, named <name>-index.tgz
+
+    :param name: name of the collection you want to import
+    :param path: path to the collection data.
+    :param method: can be 'copy', 'move' or  'link'. Copy is default.
+    """
+    path = Path(path).resolve()
+    database = path / 'pg'
+    blobs = path / 'blobs'
+    index = path / f'{name}-index.tgz'
+    log.info(f'Importing collection {name}')
+    # parsing `method`, which is not interpreted as a kwarg
+    if method != 'copy':
+        _, method = method.split('=', 1)
+        if method not in ['copy', 'move', 'link']:
+            raise RuntimeError(f'unknown method: {method}')
+
+    node_name = name.lower()
+    if name != node_name:
+        log.info(f'Renaming "{name}" to "{node_name}"')
+    collection_path = Path(config.liquid_volumes) / 'collections' / node_name
+    if collection_path.exists():
+        if confirm(f'collection {name} already exists, overwrite (hoover needs to be running)?'):
+            redeploy = True
+            purge_collection(name)
+        else:
+            log.info(f'exiting and not overwriting data for {name}')
+            exit()
+    else:
+        redeploy = False
+        (collection_path).mkdir(parents=True)
+
+    # copy the pg dir
+    database = Path(database).resolve(strict=True)
+    if not (database / 'data' / 'PG_VERSION').exists():
+        raise RuntimeError("database is not a valid Postgres database")
+    pg_src = database
+    pg_dst = collection_path / 'pg'
+    import_dir(pg_src, pg_dst, method=method)
+
+    # copy the blobs dir
+    blobs = Path(blobs).resolve(strict=True)
+    blob_src = blobs
+    blob_dst = collection_path / 'blobs'
+    import_dir(blob_src, blob_dst, method=method)
```

Comment:
I think it's safer to `tar` the blob folder to a single file. For a large collection, it will consist of a huge number of little files, it's easier to manage them as a single archive. And you can pass it through `gzip -1` to make the text chunks a bit smaller, which will make a difference, since a lot of the blobs are actually intermediate processing results in JSON format.

And please leave a blank line before you start processing the index. When you look at the screen from 5 meters away you should see 3 code chunks, for blobs, index and pg. :)

Reply:
You mean having an archive as the input of importing, unpacking it in the process and respectively creating an archive during the export?! Will do!